File: Arrays/transpose.py
from typing import List


class Solution:
    def transpose(self, mat: List[List[int]]) -> List[List[int]]:
        # Create a matrix with dimensions swapped (columns x rows)
        b = [[0] * len(mat) for _ in range(len(mat[0]))]

        for i in range(len(mat)):
            for j in range(len(mat[0])):
                b[j][i] = mat[i][j]

        return b


# --- Input handling ---
if __name__ == "__main__":
    # Take input for number of rows and columns
    rows = int(input("Enter number of rows: "))
    cols = int(input("Enter number of columns: "))

    print(f"Enter {rows * cols} elements (row-wise):")
    elements = list(map(int, input().split()))

    if len(elements) != rows * cols:
        print("Error: Incorrect number of elements!")
        exit()

    # Convert flat list to 2D matrix
    mat = [elements[i * cols:(i + 1) * cols] for i in range(rows)]

    # Create object and call method
    sol = Solution()
    result = sol.transpose(mat)

    print("Transposed Matrix:")
    for row in result:
        print(row)

File: Arrays/transpose.c
#include <stdio.h>

int main() {
    int rows, cols;

    // Input rows and columns
    printf("Enter number of rows: ");
    scanf("%d", &rows);
    printf("Enter number of columns: ");
    scanf("%d", &cols);

    int mat[100][100], transposed[100][100];

    // Input matrix elements
    printf("Enter %d elements (row-wise):\n", rows * cols);
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            scanf("%d", &mat[i][j]);
        }
    }

    // Transpose logic
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            transposed[j][i] = mat[i][j];
        }
    }

    // Output the transposed matrix
    printf("Transposed Matrix:\n");
    for (int i = 0; i < cols; i++) {
        for (int j = 0; j < rows; j++) {
            printf("%d ", transposed[i][j]);
        }
        printf("\n");
    }

    return 0;
}
